### ISSUE TITLE
Unify return format of `configure_optimizers`

### ISSUE THREAD

**Issue by wusuowei60** [PyTorchLightning/pytorch-lightning#7369]
## 🚀 Feature

Unify how optimizers and schedulers are configured in `configure_optimizers`, making the syntax more consistent.

### Motivation

Currently,, in `configure_optimizers` you can do this (code from documentation)

```python
def configure_optimizers(self):
   optimizers = [Adam(...), SGD(...)]
   schedulers = [
      {
         'scheduler': ReduceLROnPlateau(optimizers[0], ...),
         'monitor': 'metric_to_track',
         'interval': 'step',
         'frequency': 10,
         'strict': True,
      },
      LambdaLR(optimizers[1], ...)
   ]
   return optimizers, schedulers
```

or this

```python
def configure_optimizers(self):
   optimizer1 = Adam(...)
   optimizer2 = SGD(...)
   scheduler1 = ReduceLROnPlateau(optimizer1, ...)
   scheduler2 = LambdaLR(optimizer2, ...)
   return (
       {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track'},
       {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
   )
```

The first returns two lists of the same length, while the second returns a list of dicts, each dict representing an optimizer-scheduler pair.

This seems already like a duplicate, let alone in the first you use `scheduler` as key but in the second you should use `lr_scheduler`.

Besides, scheduler properties such as `interval` can only be specified in the first way, i.e.

```python
   return (
       {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track', 'interval': 'step'},
       {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
   )
```

won't work (#6948 mentioned this as well), which is very confusing and inconsistent 😕.

### Pitch

As a user, I simply stick to the first way and always returns two lists in `configure_optimizers`. But for the sake of new users, would it be feasible to **deprecate the second way and leave only the first configuring format**?

**Comment by carmocca** [comment]
> Besides, scheduler properties such as interval can only be specified in the first way, i.e. ... won't work

This is not correct. You are mixing up the keys `scheduler` and `lr_scheduler`:

- `scheduler` refers to the actual scheduler instance, for example, a `ReduceLROnPLateau` object
- `lr_scheduler` refers to a dict which contains the `scheduler` as well as any other keys associated to it

So this is correct:

**Comment by wusuowei60** [comment]
Thanks for the reply.

I know `scheduler` and `lr_scheduler` are different things. My point is:

1️⃣ It seems to me that whatever one wants to do by using dicts, like

```python
return (
    {"optimizer": optimizer1, "lr_scheduler": scheduler1},
    {"optimizer": optimizer2, "lr_scheduler": scheduler2},
)
```

one can always do with lists (or is there any counter-example🤔?), like

```python
return [optimizer1, optimizer2], [scheduler1, scheduler2]
```

This is duplication.

2️⃣ While one can set `monitor` in both ways:

`return [optimizer], [{'scheduler': scheduler, 'monitor': 'metric_to_track'}]`

or

`return {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track'}`

you can set `step` only in the former way.

This is both a duplication and inconsistency.

-------------

And my suggestion is, again, deprecating returning dicts

```python
return (
    {'optimizer': optimizer1, 'lr_scheduler': scheduler1},
    {'optimizer': optimizer2, 'lr_scheduler': scheduler2},
)
```

in favor of using lists

```python
return [optimizer1, optimizer2], [scheduler1, scheduler2]
```

**Comment by carmocca** [comment]
> return {'optimizer': optimizer1, 'lr_scheduler': scheduler1, 'monitor': 'metric_to_track'}

Where have you seen that monitor can be set this way? It does not work, so 2️⃣ is not inconsistent.

> And my suggestion is, again, deprecating returning dicts

Dict have a purpose which is the `frequency` key - the only key allowed at the root dict level other than `optimizer` and `lr_scheduler`. This can not be supported with just lists.

Check out our docs: https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#configure-optimizers, particularly the `frequency` note

**Comment by wusuowei60** [comment]
> Where have you seen that monitor can be set this way?

Right in the docs https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#learning-rate-scheduling

> Dicts have a purpose which is the frequency key...

I do missed this point, thanks for pointing out.

**Comment by carmocca** [comment]
> Right in the docs https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#learning-rate-scheduling

Okay! Thanks, I'll fix it :)

Do you have any other comments or confusing parts?

**Comment by wusuowei60** [comment]
No more on this topic.

Maybe I will try to live with more than one way to configure optimizers🤷.

**Comment by itsikad** [comment]
> Do you have any other comments or confusing parts?

Can you please elaborate what's the order of operations in the case of multiple optimizers?

```
for opt_idx:
   loss = training_step(batch. optimizer_idx=opt_idx)
   loss.backward()
   optimizer_{opt_idx}.step()
   optimizer_{opt_idx}.zero_grad()
```

OR

```
for opt_idx:
   loss = training_step(batch. optimizer_idx=opt_idx)
   loss.backward()

for opt_idx:
    optimizer_{opt_idx}.step()
    optimizer_{opt_idx}.zero_grad()
```

**Comment by carmocca** [comment]
(A). Relevant code:

https://github.com/PyTorchLightning/pytorch-lightning/blob/a2a65abf6844f7fcd49e0bfbfb7875206a8e7dd3/pytorch_lightning/trainer/training_loop.py#L692-L731
